src/deepsparse/transformers/utils/token_generator.py

```diff
@@ -77,6 +77,10 @@ def generate(self, logits: numpy.ndarray) -> numpy.ndarray:
         :param logits: the logits from the model with shape (vocab_size,)
         :return: the sampled token
         """
+        # make a copy of logits to avoid modifying the original
+        # logits distribution in-place
+        logits = logits.copy()
```

Comment:
does this only happen when we are explicitly returning logits? this seems like it could be expensive

Reply:
Good point. This will happen regardless of whether we are returning logits to the user or not. However, I do not think that it is worth making this logic more complex in favor of memory benefits. Note that those are not the logits for the full sequence being copied, but only the last logits array of shape `vocab_size`.